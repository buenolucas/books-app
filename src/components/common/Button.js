import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import Typography from './Typography';
import Theme from '../../Theme';

const ButtonContainer = styled.TouchableOpacity`
    color: ${({theme}) => theme.colors.white}
    background-color: ${({theme}) => theme.colors.buttonPrimary}
    width:100%;
    display:flex;
    alignItems:center;
    padding: ${({theme}) =>
      theme.spacing.tiny +
      theme.spacing.xTiny +
      'px ' +
      theme.spacing.small +
      'px'};
`;

const Button = props => {
  const {label, onPress} = props;
  const typograpyStyle = {color: '#FFFFFF'};

  return (
    <ButtonContainer onPress={onPress}>
      <Typography type="caption1" style={typograpyStyle}>
        {label}
      </Typography>
    </ButtonContainer>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  primary: PropTypes.bool,
};
Typography.defaultProps = {
  primary: true,
};
export default Button;
