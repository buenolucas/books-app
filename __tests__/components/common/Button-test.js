import React from 'react';
import {shallow} from 'enzyme';
import Button from '../../../src/components/common/Button';
import 'jest-styled-components';

const pressFn = jest.fn();

describe('Button', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      const component = shallow(
        <Button label="button render test" onPress={pressFn} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render with color variation', () => {
      const component = shallow(
        <Button primary={false} label="button render test" onPress={pressFn} />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
