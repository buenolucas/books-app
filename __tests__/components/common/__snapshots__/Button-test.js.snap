// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering renders without crashing 1`] = `
<Styled(TouchableOpacity)
  onPress={[MockFunction]}
>
  <Typography
    primary={true}
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
    type="caption1"
  >
    button render test
  </Typography>
</Styled(TouchableOpacity)>
`;

exports[`Button Rendering should render with color variation 1`] = `
<Styled(TouchableOpacity)
  onPress={[MockFunction]}
>
  <Typography
    primary={true}
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
    type="caption1"
  >
    button render test
  </Typography>
</Styled(TouchableOpacity)>
`;
